Show stock count on purchase button

bobrippling opened the pull request.

chrisprice commented on lib/stock.js:
```diff
@@ -18,6 +18,8 @@ const fields = [
   { header: 'Tx List (Insight)', name: 'txListUrl', parse: String },
   { header: 'Tx List (API)', name: 'txListApi', parse: String },
   { header: 'Funding Tx', name: 'fundingTx', parse: String }
+  { header: 'Sold (ScottCoin)', name: 'soldScottcoinTotal', parse: Number }
```
Nice try! Unfortunately a couple of issues -
* Those columns are identified by `N` and `P` but the range on line 5 only pulls in `A:M`
* Those fields use some horrible macro in the sheet to update once every never. Ideally we would pull it from the insight API directly (e.g. the URL specified in column `L`)

bobrippling committed:
feat: stock counting

Also filter the stock before counting to reduce the amount of GETs to
the txListApi URL.

Assumptions:
- The price of an item doesn't change. If it changes, then
  `total = amount / price` doesn't hold.

- Rounding `amount / price` gives us the correct count. Even with an
  infinite precision calculator we get non-integer totals, because
  multiples of the price can't be held in a bitcoin/scottcoin
  transaction amount. So we use Math.round() which gets us to the right
  value.

bobrippling committed:
fix: pull out stock filtering to index.js
